odxtools/cli/_print_utils.py

```diff
@@ -53,31 +63,197 @@ def print_diagnostic_service(
+        resp = service.positive_responses[0]
+        print(f"   Service Parameters: {resp.parameters}\n")
+        table = extract_parameter_tabulation_data(list(resp.parameters))
+        print(tabulate(table, headers='keys', tablefmt='presto'))
+        print(f"\n   Message format of the positive response:")
+        resp.print_message_format(
+            indent=0, allow_unknown_lengths=allow_unknown_bit_lengths, print_objects=print_objects)
+
+    # Negative Response
+    print(f"\n  [yellow]Negative Response Properties[/yellow]:")
+    print(f"   Number of Negative Responses: {len(service.negative_responses)}")
+    print(f"   Negative Responses: {service.negative_responses}")
+    if len(service.negative_responses) == 1:
+        resp = service.negative_responses[0]
+        print(f"   Service Parameters: {resp.parameters}\n")
+        table = extract_parameter_tabulation_data(list(resp.parameters))
+        print(tabulate(table, headers='keys', tablefmt='presto'))
+        print(f"\n   Message format of a negative response:")
+        resp.print_message_format(
+            indent=0, allow_unknown_lengths=allow_unknown_bit_lengths, print_objects=print_objects)
+
+    print("\n")
 
     if (service.positive_responses and
             len(service.positive_responses) > 1) or (service.negative_responses and
                                                      len(service.negative_responses) > 1):
         # Does this ever happen?
         raise NotImplementedError(
             f"The diagnostic service {service.odx_id} offers more than one response!")
+
+
+def extract_service_tabulation_data(services: List[DiagService]) -> Dict[str, Any]:
+    # extracts data of diagnostic services into Dictionary which can be printed by tabulate module
+
+    name = []
+    semantic = []
+    request: List[Optional[str]] = []
+
+    for service in services:
+        assert isinstance(service, DiagService)
+        name.append(service.short_name)
+        semantic.append(service.semantic)
+        if not getattr(service.request, "required_parameters", None):
+            if len(service().hex().upper()) > 24:
+                request.append('0x' + service().hex().upper()[:24] + '...')
+            else:
+                request.append('0x' + service().hex().upper())
+        else:
+            request.append(None)
+
+    return {'Name': name, 'Semantic': semantic, 'Hex-Request': request}
+
+
+def extract_parameter_tabulation_data(parameters: List[Parameter]) -> Dict[str, Any]:
+    # extracts data of parameters of diagnostic services into Dictionary which can be printed by tabulate module
+
+    name = []
+    byte = []
+    bit_length: List[Optional[int]] = []
+    semantic = []
+    param_type = []
+    value: List[Optional[str]] = []
+    value_type: List[Optional[str]] = []
+    data_type: List[Optional[str]] = []
+    dop: List[Optional[str]] = []
+
+    for param in parameters:
+        assert isinstance(param, Parameter)
```

Comment:
What else would it be?

Reply:
(seems to be an artifact of splitting the original function :))

Reply:
Indeed, check was removed now.